bjoern/request.c

```diff
@@ -305,6 +304,7 @@ void _initialize_request_module()
     STR_read = _Unicode_FromString("read");
     STR_seek = _Unicode_FromString("seek");
     VALUE_zero = _FromLong(0);
+    OBJECT_empty_bytesio = PyObject_CallMethodObjArgs(IO, STR_bytesio, NULL);
```

Comment:
I'm not sure we can simply reuse the same BytesIO object every time. Better create a new instance for each request

Reply:
Forgot to change it in last push ... check again the latest one. thoroughly tested with python3 for both empty body and none-empty body.